From 50f08574a04c2c7f12d2bf207fc41c6a1137c923 Mon Sep 17 00:00:00 2001
From: Roel Janssen <roel@gnu.org>
Date: Mon, 5 Feb 2018 11:25:38 +0100
Subject: [PATCH 1/2] Adapt command-line options for StrelkaPostProcess.

* lib/HMF/Pipeline/SomaticVariants.pm: Expose $tumor_bam_path to the template.
* templates/StrelkaPostProcess.sh.tt: Add -b ($tumor_bam_path) parameter.
---
 lib/HMF/Pipeline/SomaticVariants.pm | 5 +++--
 templates/StrelkaPostProcess.sh.tt  | 3 ++-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/lib/HMF/Pipeline/SomaticVariants.pm b/lib/HMF/Pipeline/SomaticVariants.pm
index a6bb5c0..6fcef21 100644
--- a/lib/perl5/site_perl/5.26.1/HMF/Pipeline/SomaticVariants.pm
+++ b/lib/perl5/site_perl/5.26.1/HMF/Pipeline/SomaticVariants.pm
@@ -39,7 +39,7 @@ sub run {
     my ($strelka_job_id, $strelka_vcf) = runStrelka($tumor_sample, $recalibrated_ref_bam, $recalibrated_tumor_bam, $joint_name, $running_jobs, $dirs, $opt);
     push @{$opt->{RUNNING_JOBS}->{somvar}}, $strelka_job_id;
 
-    my $post_process_job_id = postProcessStrelka($joint_name, $final_vcf, $strelka_job_id, $strelka_vcf, $tumor_sample, $dirs, $opt);
+    my $post_process_job_id = postProcessStrelka($joint_name, $final_vcf, $strelka_job_id, $strelka_vcf, $tumor_sample, $recalibrated_tumor_bam, $dirs, $opt);
     push @{$opt->{RUNNING_JOBS}->{somvar}}, $post_process_job_id;
 
     my $done_job_id = markDone($done_file, [ $strelka_job_id, $post_process_job_id ], $dirs, $opt);
@@ -85,7 +85,7 @@ sub runStrelka {
 }
 
 sub postProcessStrelka {
-    my ($joint_name, $final_vcf, $strelka_job_id, $strelka_vcf, $tumor_sample, $dirs, $opt) = @_;
+    my ($joint_name, $final_vcf, $strelka_job_id, $strelka_vcf, $tumor_sample, $tumor_bam_path, $dirs, $opt) = @_;
 
     say "\n### SCHEDULING STRELKA POST PROCESS ###";
 
@@ -98,6 +98,7 @@ sub postProcessStrelka {
         $dirs,
         $opt,
         tumor_sample => $tumor_sample,
+        tumor_bam_path => $tumor_bam_path,
         strelka_vcf => $strelka_vcf,
         final_vcf => $final_vcf,
         joint_name => $joint_name,
diff --git a/templates/StrelkaPostProcess.sh.tt b/templates/StrelkaPostProcess.sh.tt
index f8ca851..88deda1 100644
--- a/share/hmf-pipeline/templates/StrelkaPostProcess.sh.tt
+++ b/share/hmf-pipeline/templates/StrelkaPostProcess.sh.tt
@@ -16,7 +16,8 @@ java -Xmx[% opt.STRELKAPOSTPROCESS_MEM %]G \
     -v "$input_vcf" \
     -hc_bed "[% opt.HIGH_CONFIDENCE_BED %]" \
     -t "[% tumor_sample %]" \
-    -o "$output_vcf"
+    -o "$output_vcf" \
+    -b "[% tumor_bam_path %]"
 
 assert_not_empty "$output_vcf"
 input_vcf="$output_vcf"
-- 
2.15.0

